.pre-commit-config.yaml

```diff
@@ -294,6 +294,13 @@ repos:
         entry: "\\|\\s*safe"
         files: \.html$
         pass_filenames: true
+      - id: no-providers-in-core-examples
+        language: pygrep
+        name: No providers imports in core example DAGs
+        description: The core example DAGs have to no dependencies other than core Airflow
+        entry: "^\\s*from airflow.providers.*"
```

Comment:
Does this pattern cover cases like `from airflow.providers import ...`? Seems not to me

Reply:
Yes it works:
```
➜ pre-commit run no-providers-in-core-examples --all-files
No providers imports in core example DAGs................................Failed
- hook id: no-providers-in-core-examples
- exit code: 1

airflow/example_dags/example_complex.py:29:from airflow.providers import google
```

Reply:
Works also for not top level imports:
```
No providers imports in core example DAGs................................Failed
- hook id: no-providers-in-core-examples
- exit code: 1

airflow/example_dags/example_complex.py:35:    from airflow.providers import google
```

Reply:
The `*` is not glob pattern but a regexp definition `.*`

Reply:
But this means that the `.` between should be escaped 👀

Reply:
Cool. Thanks for the clarification!

Reply:
Haha, gotcha either here or there